react-components/src/architecture/base/reactUpdaters/CommandsUpdater.ts

```diff
@@ -0,0 +1,70 @@
+/*!
+ * Copyright 2024 Cognite AS
+ */
+
+import { type RevealRenderTarget } from '../renderTarget/RevealRenderTarget';
+
+// By calling update(), it forces the updating of all the command and tools to be done only once for
+// each user interaction, since it is done the first time the systems goes into onIdle state.
+// It is only updating the commands and tools you see in the toolbars.
+
+// The update() will typically be called when click a button, delete or add an object,
+// set object selected or visible. This will be done by the framework itself, but it may
+// be cases where you have to do it manually. Then call:
+//
+//   CommandsUpdater.update(renderTarget);
+
+export class CommandsUpdater {
+  private static _idleCallbackId: number = -1;
+  private static _renderTarget: RevealRenderTarget | undefined = undefined;
+
+  public static update(renderTarget: RevealRenderTarget | undefined): void {
+    if (renderTarget === undefined) {
+      return; // If this is not given, we do not know which target to update
+    }
+    if (this._idleCallbackId >= 0) {
+      return; // Already requested
+    }
+    this._idleCallbackId = window.requestIdleCallback(this.onIdle);
+    this._renderTarget = renderTarget;
+  }
+
+  public static dispose(): void {
+    if (this._idleCallbackId < 0) {
+      return;
+    }
+    window.cancelIdleCallback(this._idleCallbackId);
```

Comment:
missing `renderTarget` assigning to `undefined` in `dispose` which can cause a memory leak

Reply:
Thanks! See  f7be7292ab4ef1f3d9fffeda7da245a4259fc5f5